nimare/generate.py

```diff
@@ -0,0 +1,267 @@
+
+    Returns
+    -------
+    source : :obj:`dict`
+        study information in nimads format
+    """
+    source = {}
+    for sample_size, (study, study_foci) in zip(sample_sizes, foci.items()):
+        source[f"study-{study}"] = {
+            "contrasts": {
+                "1": {
+                    "coords": {
+                        "space": space,
+                        "x": [c[0] for c in study_foci],
+                        "y": [c[1] for c in study_foci],
+                        "z": [c[2] for c in study_foci],
+                    },
+                    "metadata": {
+                        "sample_sizes": [sample_size],
+                    },
+                }
+            }
+        }
+
+    return source
+
+
+def create_foci(
+    foci_num,
+    fwhm,
+    studies,
+    foci_coords=None,
+    foci_noise=0,
+    foci_weights=None,
+    seed=42,
+    space="MNI",
+):
+    """Generate study specific foci.
+
+    Parameters
+    ----------
+    foci_num : :obj:`int` or :obj:`list`
+        The number of foci to be generated per study.
+    fwhm : :obj:`float` or :obj:`list`
+        Full width at half maximum (fwhm) to define the probability
+        spread of the foci or list of fwhms identifying the fwhm
+        for each foci.
+    studies : :obj:`int`
+        Number of studies to generate.
+    foci_coords : :obj:`list` of three item array_like objects (Optional)
+        Ground truth coordinates of foci.
+        If not specified, coordinates will be randomly selected.
+        If not the same length as the number of foci (i.e., `foci_num`),
+        foci will either be over-sampled or under-sampled.
+    foci_noise : :obj:`int` or :obj:`list`
+        Number of foci considered to be noise in each study
+        or a list of integers representing how many noise focis
+        there should be in each study.
+    foci_weights : :obj:`list`
+        Weighing of each foci representing the probability of
+        that foci being sampled in a study.
+    seed : :obj:`int` (Optional)
+        Random state to reproducibly initialize random numbers.
+    space : :obj:`str` (Default="MNI")
+        The template space the coordinates are reported in.
+
+    Returns
+    -------
+    ground_truth_foci : :obj:`list`
+        List of 3-item tuples containing x, y, z coordinates
+        of the ground truth foci.
+    foci_dict : :obj:`dict`
+        Dictionary with keys representing the ground truth foci, and
+        whose values represent the study specific foci.
+    """
+    rng = np.random.RandomState(seed=seed)
+
+    if space == "MNI":
+        template_img = nib.load(
+            os.path.join(get_resource_path(), "templates", "MNI152_2x2x2_brainmask.nii.gz")
+        )
+    else:
+        raise NotImplementedError("Only coordinates for the MNI atlas has been defined")
+
+    # use a template to find all "valid" coordinates
+    template_data = template_img.get_data()
```

Comment:
```suggestion
    template_data = template_img.get_fdata()
```
`get_data` is deprecated.

Reply:
thanks! incorporated this change.